### PR TITLE
Ports over changes from my remaining odometry branch

### PR DESCRIPTION
**Pull request by KipHamiltons**
#453 has some good stuff. This moves most of it over.

### CODE REVIEW

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 1]
```diff
@@ -369,15 +370,15 @@ namespace module::input {
                             NUClear::log<NUClear::WARN>(makeErrorString("Right FSR", input.fsr.right.error_flags));
                         }
 
-                        // Read through all of our sensors
-                        for (uint32_t i = 0; i < 20; ++i) {
-                            auto& original = utility::platform::getRawServo(i, input);
+                        // Loop through all the servos, emitting a message::input::Sensors for each one
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:373]: Is "emitting" the write word for this?

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 2]
```diff
@@ -495,7 +497,9 @@ namespace module::input {
                          *                  Kinematics                  *
                          ************************************************/
 
+                        // Htx is a Map from ServoID to (a homogeneous transform from that joint to the torso)
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:500]: What?

**Review by Bidski — CHANGES_REQUESTED** [review]
Brief review for now

File: module/input/SensorFilter/src/SensorFilter.cpp (hunk 1 above)
Comment (KipHamiltons) [reply module/input/SensorFilter/src/SensorFilter.cpp:373]: No it's not. good catch.

File: module/input/SensorFilter/src/SensorFilter.cpp (hunk 2 above)
Comment (KipHamiltons) [reply module/input/SensorFilter/src/SensorFilter.cpp:500]: I was having trouble understanding what it was, and once I found out I added this comment, but I agree that my wording is awkward, although I think it's accurate. Do you have a better explanation of `Htx`? I struggled to get better words for it without being _that_ literal

Comment (Bidski) [reply module/input/SensorFilter/src/SensorFilter.cpp:499]: "Htx is a map from ServoID to homogeneous transforms from each ServoID to the torso", or something like that?

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 3]
```diff
@@ -564,125 +571,133 @@ namespace module::input {
                                              MeasurementType::GYROSCOPE());
 
                         // Calculate accelerometer noise factor
-                        Eigen::Matrix3d acc_noise =
+                        const Eigen::Matrix3d acc_noise =
                             config.motionFilter.noise.measurement.accelerometer
+                            // Add noise which is proportional to the square of how much we are moving, minus gravity
+                            // This means that the faster we move, the noisier we think the measurements are
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:577]: "faster we move" is not really a phrase that matches with acceleration (large acceleration in negative direction could mean we are slowing down for instance)

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 4]
```diff
@@ -564,125 +571,133 @@ namespace module::input {
...
+                                // instead stores the position that foot would be if/when it becomes flat on the ground
                                 Eigen::Affine3d Htg(utility::motion::kinematics::calculateGroundSpace(Htf, Hwt));
 
                                 footlanding_Hwf[side]                   = Hwt * Htg;
                                 footlanding_Hwf[side].translation().z() = 0.0;
 
+                                // Store the current foot down state for next time
                                 previous_foot_down[side] = true;
                             }
+                            // Else is down, and didn't hit the ground this time step
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:612]:
```suggestion
                            // This sides foot is down, but it didn't hit the ground in this time step
```

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 5]
```diff
@@ -564,125 +571,133 @@ namespace module::input {
...
 
                                 // do a foot based orientation update
-                                Eigen::Quaterniond Rwt(footlanding_Hwt.linear());
+                                const Eigen::Quaterniond Rwt(footlanding_Hwt.linear());
                                 motionFilter.measure(Rwt.coeffs(),
                                                      config.motionFilter.noise.measurement.flatFootOrientation,
                                                      MeasurementType::FLAT_FOOT_ORIENTATION());
                             }
+                            // Else the foot is off the ground, so we make sure that for the next time step, we know
+                            // that this time step, the foot was off the ground
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:629]:
```suggestion
                            // Otherwise this sides foot is off the ground, so we make sure that for the next time step, we know
                            // that this time step, the foot was off the ground
```

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 6]
```diff
@@ -694,16 +709,6 @@ namespace module::input {
                             // createRotationZ : Mat size [3x3]
                             // Rwt : Mat size [3x3]
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:710]: Are these comments still relevant?

**Review by Bidski — CHANGES_REQUESTED** [review]

File: module/input/SensorFilter/src/SensorFilter.cpp (hunk 6 above)
Comment (KipHamiltons) [reply module/input/SensorFilter/src/SensorFilter.cpp:710]: nope. I have deleted them on the next commit :+1:

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 7]
```diff
@@ -564,146 +571,141 @@ namespace module::input {
...
 
                         // Accelerometer measurement update
                         motionFilter.measure(sensors->accelerometer, acc_noise, MeasurementType::ACCELEROMETER());
-
-                        for (auto& side : {BodySide::LEFT, BodySide::RIGHT}) {
-                            bool foot_down      = sensors->feet[side].down;
-                            bool prev_foot_down = previous_foot_down[side];
-                            Eigen::Affine3d Htf(
+                        // This loop calculates the Hwf transform for feet if they have just hit the ground. If they
+                        // have not just hit the ground, it uses the previous Hwf value. This assumes that once the foot
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:585]: i.e, we assume our foot cant slip (make our assumptions explicit)

File: module/platform/Webots/src/Webots.cpp [hunk 8]
```diff
@@ -441,12 +441,8 @@ namespace module::platform {
                     const uint32_t sensor_timestep = std::max(min_sensor_time_step, time_step);
                     const uint32_t camera_timestep = std::max(min_camera_time_step, time_step);
 
-                    // Construct the ActuatorRequests message
-                    ActuatorRequests actuator_requests = create_sensor_time_steps(sensor_timestep, camera_timestep);
-                    for (auto& servo : servo_state) {
-                        if (servo.dirty) {
-                            // Servo is no longer dirty
-                            servo.dirty = false;
+                        // Create servo position message
```
Comment (Bidski) [inline module/platform/Webots/src/Webots.cpp:444]: Why is this here?

**Review by Bidski — CHANGES_REQUESTED** [review]

**Comment by KipHamiltons** [comment]
unrelated changes have been purged from this branch as well :grimacing: :joy:

File: module/input/SensorFilter/src/SensorFilter.cpp (hunk 7 above)
Comment (KipHamiltons) [reply module/input/SensorFilter/src/SensorFilter.cpp:585]: I have added an extra clause to make it more explicit :+1:

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 9]
```diff
@@ -564,146 +571,141 @@ namespace module::input {
...
+                                // instead stores the position that foot would be if/when it becomes flat on the ground
                                 Eigen::Affine3d Htg(utility::motion::kinematics::calculateGroundSpace(Htf, Hwt));
 
                                 footlanding_Hwf[side]                   = Hwt * Htg;
                                 footlanding_Hwf[side].translation().z() = 0.0;
 
+                                // Store the current foot down state for next time
                                 previous_foot_down[side] = true;
                             }
+                            // This sides foot is down, but it didn't hit the ground in
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:612]: in what?

**Review by Bidski — CHANGES_REQUESTED** [review]

**Comment by KipHamiltons** [comment]
Superseded by #557
